File: internal/analyzer/testdata/python/bad_workflow.py
from temporalio import workflow
import datetime
import time
import random
import requests
import threading
import queue


@workflow.defn
class BadWorkflow:
    @workflow.run
    async def run(self, input: str) -> str:
        now = datetime.datetime.now()
        time.sleep(1)
        val = random.randint(1, 10)
        resp = requests.get("http://example.com")
        t = threading.Thread(target=lambda: None)
        lock = threading.Lock()
        q = queue.Queue()
        return str(val)

File: internal/analyzer/testdata/java/BadWorkflow.java
package com.example.workflows;

import io.temporal.workflow.WorkflowInterface;
import io.temporal.workflow.WorkflowMethod;
import java.time.Instant;
import java.util.Date;
import java.util.Random;

@WorkflowInterface
public interface BadWorkflowInterface {
    @WorkflowMethod
    String run(String input);
}

public class BadWorkflow implements BadWorkflowInterface {
    @WorkflowMethod
    public String run(String input) {
        long now = System.currentTimeMillis();
        Thread.sleep(1000);
        double r = Math.random();
        new Random().nextInt();
        new Thread(() -> {}).start();
        new ReentrantLock();
        return "done";
    }
}
